### PR TITLE
store/tikv: use resolve lock lite for snapshot.Get()

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Problem Summary:

If a large transaction runs into 2PC commit, at the same time, another transaction read and conflict, the resolve lock process takes too long.

image

Note the QPS drop to 0 at 05:25, that is because the resolve lock takes too long.

image

### What is changed and how it works?

What's Changed:

How it Works:

Use resolve lock lite instead of resolve region for snapshot.Get().
Point get request would benefit from this and do not block too long.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test
- No code

Side effects

### Release note 

- No release note

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @lysu @youjiali1995

File: ddl/column.go [hunk 1]
```diff
@@ -165,11 +165,11 @@ func onAddColumn(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, err error)
 		return ver, nil
 	}
 
-	failpoint.Inject("errorBeforeDecodeArgs", func(val failpoint.Value) {
+	if val, _err_ := failpoint.Eval(_curpkg_("errorBeforeDecodeArgs")); _err_ == nil {
```
Comment (lysu) [inline ddl/column.go:168]: need `make failpoint-disable` - -

Comment (tiancaiamao) [reply ddl/column.go]: Sorry, I run `make failpoint-enable` and commit and forgot to reset those changes.

**Comment by tiancaiamao** [comment]
DATA RACE found in `failpoint.Enable`  https://github.com/pingcap/tidb/issues/17389

/rebuild

**Comment by tiancaiamao** [comment]
TestCoprocessorStreamRecvTimeout https://github.com/pingcap/tidb/issues/16958

/rebuild

**Comment by tiancaiamao** [comment]
DATA RACE found in `ast.(*ColumnNameExpr).Accept()`
Just now fixed by https://github.com/pingcap/tidb/pull/17353

**Comment by tiancaiamao** [comment]
Fix https://github.com/pingcap/tidb/issues/16963

**Pull request merged**